### ISSUE TITLE
[java] Lombok getter annotation on enum is not recognized correctly

### ISSUE THREAD

**Issue by mmoehring** [pmd/pmd#527]
**Rule Set:**
`<ruleref="rulesets/java/design.xml"/>`

**Description:**
A `@Getter` annotation from Lombok on an enum is not recognized by PMD. The field for which the getter is generated is flagged with the SingularFieldRule.
Setting the `@Getter` directly on the field works.

**Code Sample demonstrating the issue:**
```java
@Getter
public enum Foo {
    BAR(1);

    private final int number;

    Foo(final int number) {
        this.number = number;
    }
}
```

**Running PMD through:** *[Gradle]*

**Comment by jsotuyod** [comment]
Thanks for the report. you are absolutely right, the same will happen to all Lombok aware rules when dealing with enums... We are currently only checking in classes / interfaces, but not on enums. We should have [the same check](https://github.com/pmd/pmd/blob/master/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractLombokAwareRule.java#L57-L58) for those nodes of the AST too.

Should be simple enough to fix.

**Comment by jsotuyod** [comment]
This has been fixed thanks to @oowekyala. To be included in the next PMD release
